### ISSUE TITLE
Reading plans: re-design / improve screen / toolbar UI/UX

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#284]
Ideas:
 - Re-design toolbar (remove commentary/bible buttons), show something useful instead
 - Remove big "done" button, instead add "up" arrow to toolbar (like in speak screen for example)
 - Consider implementing speak feature better (now uses legacy speak system)

**Comment by tuomas2** [comment]
Reading plan names (that are distributed with And Bible) should be also translatable similar
way as other strings (they should be in values/strings.xml).

**Comment by timbze** [comment]
I didn't think about that right now. I was thinking of adding comment lines with the # symbol in the beginning of properties files. The first line with # sign would be Plan name, and the following lines would be description. M'Cheyne properties file start after my modifications is like this:
# M'Cheyne 1-Year 4-chapter/day
# This plan is based on the M’Cheyne reading system, featuring four different readings
# for use in both family and personal devotions. Each day has two passages from the Old Testament,
# one from the New Testament, and one from either the Psalms or the Gospels. In one year,
# you read the Old Testament once and the New Testament and Psalms twice.

Because of user defined plans, we would either need to do something like that, or we could have a screen where they can add the name for their plan, then the plan information, to add it. I think it would be better to have in the file, that way it would be easier to distribute the file to my friends so we could all do my custom plan.

**Comment by timbze** [comment]
Forgot to say, we can have strings that can be translated for the names of the plans that are distributed with AndBible, and User defined plans don't need translations.

**Comment by mjdenham** [comment]
Reading from the files every time actually works quite well and gives no problems.
For plan status to be backed up you only need to save the current state to the db instead of saving to preferences.  
If you remove the Bible and Commentary buttons from the toolbar you will need to provide another way to navigate to the Bible and Commentary.
The significance of the Done button is that it marks the days reading plan as completed.  An up-arrow would be confusing.

**Comment by mjdenham** [comment]
Reading from the files every time actually works quite well and gives no problems.
For plan status to be backed up you only need to save the current state to the db instead of saving to preferences.  
If you remove the Bible and Commentary buttons from the toolbar you will need to provide another way to navigate to the Bible and Commentary.
The significance of the Done button is that it marks the days reading plan as completed.  An up-arrow would be confusing.
Think through this carefully as this is a widely used feature and you could well upset a lot of people if you make a bad call.
Saving preferences to the db would be a safe and useful start that would also help to familiarise you with the architecture.

**Comment by timbze** [comment]
@mjdenham Thanks for your ideas.

> For plan status to be backed up you only need to save the current state to the db instead of saving to preferences.

You are right this could be done. I was thinking it would be better practice and faster to just get the days needed from DB instead of loading the entire plan into memory every time, but we are not working with very big files so the difference would not be huge.
Another reason I wanted to do this is if users have their own custom plans, that they will be loaded into DB in their entirety, that way if SD card is lost or erased the plan will continue functioning normally.

> If you remove the Bible and Commentary buttons from the toolbar you will need to provide another way to navigate to the Bible and Commentary.

I do think adding the "up" arrow on top left might be a good alternative to those 2 buttons, that would bring the user back to Main Bible screen. We would just need to continue to have a simple and intuitive way of changing either the plan or the day by the user.

> The significance of the Done button is that it marks the days reading plan as completed. An up-arrow would be confusing.

I agree that there has to be a clean way of going to the next day. If we implement the reading plan by-date #292, I believe it will become necessary to be able to go to previous and next days easier, and if we are able to figure out a good design for that, we could then remove the done button. I would, however, work on this functionality after adding plans to DB if we end up doing that.

Any further thoughts you have on my thoughts are highly welcomed :)

**Comment by tuomas2** [comment]
> Forgot to say, we can have strings that can be translated for the names of the plans that are distributed with AndBible, and User defined plans don't need translations.

Yes I agree with this. Just the built-in ones would be nice to have translations.

**Comment by mjdenham** [comment]
I may well find prev/next reading display already in the code (I think it might have been via swiping) commented out because I originally implemented it but then decided to navigate purely by Done.

**Comment by mjdenham** [comment]
Another thing which would merit attention is the wording and method of adjusting the reading plans e.g. if you fall behind or want to change start date etc.  The current method dsecribed in the wiki is confusing and could be simplified.
